ext/Client/VisualEnvironmentHandler.lua

```diff
@@ -0,0 +1,409 @@
+	s_Object.entity:FireEvent("Disable")
+	s_Object.entity:FireEvent("Enable")
+end
+
+---@param p_ID string
+---@param p_Visibility number
+function VisualEnvironmentHandler:SetVisibility(p_ID, p_Visibility)
+	---@type VisualEnvironmentObject
+	local s_Object = self._VisualEnvironmentObjects[p_ID]
+
+	if not s_Object.entity then
+		self:InitializeVE(p_ID, p_Visibility)
+	elseif p_Visibility <= 0.0 then
+		self:DestroyVE(p_ID)
+	else
+		s_Object.ve.visibility = p_Visibility
+		local s_State = VisualEnvironmentEntity(s_Object.entity).state
+
+		if s_State then
+			s_State.visibility = p_Visibility
+			VisualEnvironmentManager:SetDirty(true)
+		else
+			self:Reload(p_ID)
+		end
+	end
+
+	if s_Object.rawPreset["RuntimeEntities"] ~= nil then
+		if p_Visibility > 0.5 then
+			m_RuntimeEntityHandler:SetVisibility(s_Object, false)
+		else
+			m_RuntimeEntityHandler:SetVisibility(s_Object, true)
+		end
+	end
+end
+
+---@param p_ID string
+---@param p_VisibilityStart number|nil
+---@param p_VisibilityEnd number
+---@param p_FadeTime number time of the transition in miliseconds
+---@param p_TransitionType EasingTransitions|nil
+function VisualEnvironmentHandler:FadeTo(p_ID, p_VisibilityStart, p_VisibilityEnd, p_FadeTime, p_TransitionType)
+	---@type VisualEnvironmentObject
+	local s_Object = self._VisualEnvironmentObjects[p_ID]
+	local s_TransitionFunction = m_Easing[p_TransitionType]
+
+	if not s_TransitionFunction then
+		-- default to linear
+		s_TransitionFunction = m_Easing["linear"]
+	end
+
+	---@type LerpProperties
+	local s_LerpProperties = {
+		enabled = true,
+		transitionFunctionName = p_TransitionType or "linear",
+		transitionFunction = s_TransitionFunction,
+		transitionTime = p_FadeTime,
+		startTime = SharedUtils:GetTimeMS(),
+		startValue = p_VisibilityStart or s_Object.ve.visibility,
+		endValue = p_VisibilityEnd,
+		firstRun = true
+	}
+
+	self._Lerping[p_ID] = s_LerpProperties
+end
+
+---@param p_ID string
+---@param p_PulseTime number time of the transition in miliseconds
+---@param p_DecreaseFirst boolean sets if the first pulse decreases the current value until 0
+---@param p_TransitionType EasingTransitions|nil
+function VisualEnvironmentHandler:Pulse(p_ID, p_PulseTime, p_DecreaseFirst, p_TransitionType)
+	---@type VisualEnvironmentObject
+	local s_Object = self._VisualEnvironmentObjects[p_ID]
+	local s_TransitionFunction = m_Easing[p_TransitionType]
+	local s_VisibilityStart
+	local s_VisibilityEnd
+
+	if s_Object.entity and s_Object.entity.state then
+		s_VisibilityStart = s_Object.entity.state.visibility
+	else
+		s_VisibilityStart = 0
+	end
+
+	if not s_TransitionFunction then
+		-- default to linear
+		s_TransitionFunction = m_Easing["linear"]
+	end
+
+	if p_DecreaseFirst then
+		s_VisibilityEnd = 0
+	else
+		s_VisibilityEnd = 1
+	end
+
+	---@type LerpProperties
+	local s_LerpProperties = {
+		enabled = true,
+		pulsing = true,
+		transitionFunctionName = p_TransitionType or "linear",
+		transitionFunction = s_TransitionFunction,
```

Comment:
Can be simplified to `transitionFunction = s_TransitionFunction or m_Easing["linear"]`